app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCapturePresenterImpl.java

```diff
@@ -693,48 +695,72 @@ public void unsupportedRuleAction() {
                 assignedValueChanged = true;
             }
         }
     }
 
     @Override
-    public void setDisplayKeyValue(String label, String value) {
+    public void setDisplayKeyValue(@NonNull String label, @NonNull String value) {
         //TODO: Implement Indicator tabs to show this field
     }
 
     @Override
-    public void setHideSection(String sectionUid) {
+    public void setHideSection(@NonNull String sectionUid) {
         if (!sectionsToHide.contains(sectionUid))
             sectionsToHide.add(sectionUid);
     }
 
     @Override
-    public void setMessageOnComplete(String message, boolean canComplete) {
+    public void setMessageOnComplete(@NonNull String message, boolean canComplete) {
         this.canComplete = canComplete;
         this.completeMessage = message;
     }
 
     @Override
-    public void setHideProgramStage(String programStageUid) {
+    public void setHideProgramStage(@NonNull String programStageUid) {
         //do not apply
     }
 
     @Override
-    public void setOptionToHide(String optionUid) {
-        optionsToHide.add(optionUid);
+    public void setOptionToHide(@NonNull String optionUid, @NonNull String field) {
+        if (!optionsToHide.containsKey(field)) {
+            optionsToHide.put(field, new ArrayList<>());
+        }
+        optionsToHide.get(field).add(optionUid);
+        StoreResult result = valueStore.deleteOptionValueIfSelected(field, optionUid);
+        if (result.component2() == ValueStoreImpl.ValueStoreResult.VALUE_CHANGED) {
+            assignedValueChanged = true;
+        }
     }
 
     @Override
-    public void setOptionGroupToHide(String optionGroupUid, boolean toHide, String field) {
-        if (toHide)
-            optionsGroupsToHide.add(optionGroupUid);
-        else if (!optionsGroupsToHide.contains(optionGroupUid))//When combined with show option group the hide option group takes precedence.
-            if (optionsGroupToShow.get(field) != null)
+    public void setOptionGroupToHide(@NonNull String optionGroupUid, boolean toHide, @NonNull String field) {
+        if (toHide) {
+            if(!optionsGroupsToHide.containsKey(field)){
+                optionsGroupsToHide.put(field,new ArrayList<>());
+            }
+            optionsGroupsToHide.get(field).add(optionGroupUid);
+            if (!optionsToHide.containsKey(field)) {
+                optionsToHide.put(field, new ArrayList<>());
+            }
+            optionsToHide.get(field).addAll(eventCaptureRepository.getOptionsFromGroups(Collections.singletonList(optionGroupUid)));
+            StoreResult result = valueStore.deleteOptionValueIfSelectedInGroup(field, optionGroupUid,true);
+            if (result.component2() == ValueStoreImpl.ValueStoreResult.VALUE_CHANGED) {
+                assignedValueChanged = true;
+            }
+        } else if (!optionsGroupsToHide.containsKey(field) || !optionsGroupsToHide.get(field).contains(optionGroupUid)) {
```

Comment:
I think eventCapturePresenter has a unit test class, if we can add this :+1:

Reply:
Nope it has not. I'll create one